src/pages/SignupPage.tsx

```diff
@@ -0,0 +1,363 @@
+import TextField from "@mui/material/TextField";
+import IconButton from "@mui/material/IconButton";
+import Button from "@mui/material/Button";
+import { AppDispatchType, RootState } from "../redux/Store";
+import React from "react";
+import CircularProgress from "@mui/material/CircularProgress";
+import Box from "@mui/material/Box";
+import { useState, FocusEvent } from "react";
+import InputLabel from "@mui/material/InputLabel";
+import { styled } from "@mui/material/styles";
+import OutlinedInput from "@mui/material/OutlinedInput";
+import FormControl from "@mui/material/FormControl";
+import FormHelperText from "@mui/material/FormHelperText";
+import InputAdornment from "@mui/material/InputAdornment";
+import Visibility from "@mui/icons-material/Visibility";
+import VisibilityOff from "@mui/icons-material/VisibilityOff";
+import CloudUploadIcon from "@mui/icons-material/CloudUpload";
+import { useNavigate } from "react-router-dom";
+import { RegistrationFormData, SaveUserData } from "../redux/RegisterSlice";
+import { useDispatch, useSelector } from "react-redux";
+
+const VisuallyHiddenInput = styled("input")({
+  clip: "rect(0 0 0 0)",
+  clipPath: "inset(50%)",
+  height: 1,
+  overflow: "hidden",
+  position: "absolute",
+  bottom: 0,
+  left: 0,
+  whiteSpace: "nowrap",
+  width: 1,
+});
+
+interface registerDataType {
+  fname: string;
+  lname: string;
+  phoneNumber: string | number;
+  email: string;
+  password: string;
+  picture: File | null;
+  confirmPassword: string;
+}
+const SignupPage: React.FC = () => {
+  const [alreadyUser, setAlreadyUser] = useState(false);
+  const [showPassword, setShowPassword] = useState({
+    password: false,
+    confirmPassword: false,
+  });
+  const [registerData, setResgisterData] = useState<registerDataType>({
+    fname: "",
+    lname: "",
+    phoneNumber: "",
+    email: "",
+    password: "",
+    picture: null,
+    confirmPassword: "",
+  });
+  const [inputFieldError, setInputFieldError] = useState({
+    firstNameError: false,
+    lastNameError: false,
+    phoneNumberError: false,
+    emailInputError: false,
+    passwordError: false,
+    confirmPasswordError: false,
+  });
+  const passwordRegex = /^(?=.*[A-Z])(?=.*[a-z])(?=.*\d)(?=.*[!@#$%^&*]).{8,}$/;
+  let fname = false;
+  let lname = false;
+  let emailInputError = false;
+  let passError = false;
+  let confirmPassword = false;
+  let numError = false;
+
+  const handleBlur = (event: FocusEvent<HTMLInputElement>) => {
+    if (event.target.id === "fname") {
```

Comment:
can you use this https://formik.org/ to validate the form @Rishab-innovative 
Your current method is not good